packages/webpack.vue.config.js

```diff
@@ -37,16 +38,14 @@ const createVueEntries = (targetPattern = 'index.ts') => {
 }
 
 /** @type WebpackConfig */
-const vueConfig = {
-    ...baseConfig,
+const vueConfig = merge(baseConfig, {
     name: 'vue',
     target: 'web',
     output: {
-        ...baseConfig.output,
         libraryTarget: 'this',
     },
     module: {
-        rules: (baseConfig.module?.rules ?? []).concat(
+        rules: [
```

Comment:
webpack-merge has its own semantics and "magic", so it's less obvious than the old code. If we suspect that the old `(baseConfig.module?.rules ?? []` approach could have subtle bugs then it's probably worth taking on a dependency.